torchaudio/functional/functional.py

```diff
@@ -1155,14 +1159,27 @@ def spectral_centroid(
         n_fft (int): Size of FFT
         hop_length (int): Length of hop between STFT windows
         win_length (int): Window size
+        min_freq (float, optional): Specify a minimum frequency to include in centroid calculation
+        max_freq (float, optional): Specify a maximum frequency to include in centroid calculation
 
     Returns:
         Tensor: Dimension `(..., time)`
     """
+    nyquist = sample_rate // 2
+    fft_bins = 1 + n_fft // 2
     specgram = spectrogram(waveform, pad=pad, window=window, n_fft=n_fft, hop_length=hop_length,
                            win_length=win_length, power=1., normalized=False)
     freqs = torch.linspace(0, sample_rate // 2, steps=1 + n_fft // 2,
                            device=specgram.device).reshape((-1, 1))
+    
+    min_freq_index = int(round((min_freq * fft_bins) / nyquist)) if min_freq is not None else 0
+    max_freq_index = int(round((max_freq * fft_bins) / nyquist)) if max_freq is not None else fft_bins
```

Comment:
IIRC TorchScript does not handle ternary operator. Can you run `pytest test/torchaudio_unittest/transforms/torchscript_consistency_cpu_test.py -k Centroid` and see if it works?

Reply:
Hi, this test seems to pass on my system. Should I still swap out the ternary?